src/compiler/checker.ts

```diff
@@ -8952,7 +8971,7 @@ namespace ts {
                 if (isRightSideOfQualifiedNameOrPropertyAccess(location)) {
                     location = location.parent;
                 }
-                if (isExpression(location) && !isAssignmentTarget(location)) {
+                if (isPartOfExpression(location) && !isAssignmentTarget(location)) {

```

Comment:
This is a bug that somehow got introduced during some refactoring for the new emitter. The old `isExpression` was renamed to `isPartOfExpression` and a new `isExpression` was introduced that doesn't look at the context (i.e. doesn't access the `parent` property).